guava/src/com/google/common/reflect/Invokable.java

```diff
@@ -203,9 +205,15 @@ abstract Object invokeInternal(@Nullable Object receiver, Object[] args)
     }
 
     @Override
-    final Object invokeInternal(@Nullable Object receiver, Object[] args)
+    final @Nullable Object invokeInternal(@Nullable Object receiver, Object[] args)
         throws InvocationTargetException, IllegalAccessException {
-      return method.invoke(receiver, args);
+      /*
+       * If the method is an instance method, invoke() will throw NullPointerException. That's the
+       * best we can do without having separate user-visible types for static and instance methods.
+       */
+      @SuppressWarnings("nullness")
+      Object result = method.invoke(receiver, args);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)